### PR TITLE
[Impeller] Replace FML_OS_PHYSICAL_IOS compile check with runtime capabilties check based on metal GPU family.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Removes all usage of FML_OS_PHYSICAL_IOS. Instead, we create a new capabilities check and set this in the metal backend based on the metal GPU family. This allows programmable blending to work on M1 and M2 devices in additon to iOS devices.

The entity_pass logic that relied on FML_OS_PHYSICAL_IOS checks has been refactored to track whether an advanced blend or backdrop filter was used separately. This allows us to determine whether or not to end the pass when we have access to the device capabilities. I could have turned the compile time checks into runtime checks, but then I'd need to plumb device capabilities into the Canvas.

Fixes https://github.com/flutter/flutter/issues/121886
Also part of https://github.com/flutter/flutter/issues/120223 by enabling this on some macOS machines

### CODE REVIEW

**Comment by jonahwilliams** [comment]
xref: https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf

**Comment by jonahwilliams** [comment]
This is called programmable blending, which looks like it is only supported on "Apple" GPU families

**Comment by jonahwilliams** [comment]
OK, I've got runtime checks working correctly, and confirmed we can use programmable blending on M1 and M2 devices. Yay!

File: impeller/entity/contents/framebuffer_blend_context.h [hunk 1]
```diff
@@ -0,0 +1,295 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
```
Comment (jonahwilliams) [inline impeller/entity/contents/framebuffer_blend_context.h:5]: Now that there are no compile time checks I could just move this back into content_context.h

File: impeller/renderer/device_capabilities.h [hunk 2]
```diff
@@ -23,6 +23,8 @@ class IDeviceCapabilities {
 
   bool SupportsTextureToTextureBlits() const;
 
+  bool SupportsFramebufferBlending() const;
```
Comment (chinmaygarde) [inline impeller/renderer/device_capabilities.h:26]: `SupportsFramebufferFetch` maybe? That we use it for blending perhaps is not related to the feature being supported.

**Review by chinmaygarde — APPROVED** [review]
Thank you! LGTM with a minor nit on naming.

File: impeller/renderer/device_capabilities.h (hunk 2 above)
Comment (chinmaygarde) [reply impeller/renderer/device_capabilities.h:26]: I just saw you comments on what Apple calls it. SGTM.

Comment (chinmaygarde) [reply impeller/renderer/device_capabilities.h:26]: My vote if for framebuffer fetch.

**Comment by jonahwilliams** [comment]
I think the iOS simulator might be lying about its support for this API 😆

File: impeller/renderer/device_capabilities.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/device_capabilities.h]: Done

**Comment by chinmaygarde** [comment]
So, its simulating the device in the response for support but not actually supporting it :/

**Comment by jonahwilliams** [comment]
Or I wonder if we're picking up the real macOS metal device but the simulator itself has different shims or validation layers? 🤷‍♂️

**Pull request merged**